### PR TITLE
Updates for Rosetta needs

### PR DESCRIPTION
**Pull request by bisgardo**
## Purpose

Ensure that the SDK satisfy the needs of Rosetta.

## Changes

* Derive `Display` on `AmountFraction`.
* Add comment to explain how to determine delegation pool of account referenced in `PaydayAccountReward`.
* Fix deserialization of `RewardsDataRaw`.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: src/types/mod.rs [hunk 1]
```diff
@@ -612,6 +612,8 @@ pub enum SpecialTransactionOutcome {
         development_charge: Amount,
     },
     /// Payment for a particular account.
+    /// The account delegates to the pool given by the last PaydayPoolReward
+    /// outcome listed before this outcome.
```
Comment (abizjak) [inline src/types/mod.rs:616]: This comment is not really in the correct place. This variant on its own has no notion of "last outcome before". That is only given meaning when these outcomes are part of the block summary.

[the remark this answers is not in the document]
Comment (abizjak) [reply src/types/mod.rs:616]: You mean like adding an `Option<BakerId>` field and construct it during deserialization of the block summary?

[the remark this answers is not in the document]
Comment (abizjak) [reply src/types/mod.rs:616]: I can add this, but it is a bit complicated since it's stateful.

[the remark this answers is not in the document]
Comment (bisgardo) [reply src/types/mod.rs:617]: Yes. I don't know if it's worth the complexity, but it would be nicer to use.

Comment (bisgardo) [reply src/types/mod.rs]: Done.

**Review by abizjak — APPROVED** [review]

**Review by td202 — APPROVED** [review]

**Pull request merged**